tsdb/engine/tsm1/engine.go

```diff
@@ -1161,6 +1161,12 @@ func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, aggregate bo
 				return err
 			}
 
+			if opt.Limit > 0 || opt.Offset > 0 || opt.SLimit > 0 || opt.SOffset > 0 {
```

Comment:
I think this is still needed even when these conditions aren't met. Otherwise, the output is still non-deterministic for queries without a limit.